Assets/Python/RiseAndFall.py

```diff
@@ -1955,92 +1717,80 @@ def warOnSpawn(self, iPlayer, lEnemies):
 
-			if (len(cityList)):
-				result = cityList[0]
-				rndNum = gc.getGame().getSorenRandNum(len(cityList), 'random city')
-				result = cityList[rndNum]
-				if (result):
-					tCityPlot = (result.getX(), result.getY())
+			if cityList:
+				city = utils.getRandomEntry(cityList)
+				if city:
+					tCityPlot = (city.getX(), city.getY())
 					tPlot = self.findSeaPlots(tCityPlot, 1, iCiv)
-					if (tPlot):
-						gc.getPlayer(iCiv).initUnit(iGalley, tPlot[0], tPlot[1], UnitAITypes.UNITAI_ASSAULT_SEA, DirectionTypes.DIRECTION_SOUTH)
-						if (teamCiv.isHasTech(iCivilService)):
-							if (iCiv == iVikings):
-								gc.getPlayer(iCiv).initUnit(iVikingHuscarl, tPlot[0], tPlot[1], UnitAITypes.UNITAI_ATTACK, DirectionTypes.DIRECTION_SOUTH)
-								gc.getPlayer(iCiv).initUnit(iVikingHuscarl, tPlot[0], tPlot[1], UnitAITypes.UNITAI_ATTACK_CITY, DirectionTypes.DIRECTION_SOUTH)
+					if tPlot:
+						x, y = tPlot
+						gc.getPlayer(iCiv).initUnit(iGalley, x, y, UnitAITypes.UNITAI_ASSAULT_SEA, DirectionTypes.DIRECTION_SOUTH)
+						if teamCiv.isHasTech(iCivilService):
+							if iCiv == iVikings:
+								gc.getPlayer(iCiv).initUnit(iVikingHuscarl, x, y, UnitAITypes.UNITAI_ATTACK, DirectionTypes.DIRECTION_SOUTH)
+								gc.getPlayer(iCiv).initUnit(iVikingHuscarl, x, y, UnitAITypes.UNITAI_ATTACK_CITY, DirectionTypes.DIRECTION_SOUTH)
 							else:
-								gc.getPlayer(iCiv).initUnit(iHeavySwordsman, tPlot[0], tPlot[1], UnitAITypes.UNITAI_ATTACK, DirectionTypes.DIRECTION_SOUTH)
-								gc.getPlayer(iCiv).initUnit(iHeavySwordsman, tPlot[0], tPlot[1], UnitAITypes.UNITAI_ATTACK_CITY, DirectionTypes.DIRECTION_SOUTH)
+								gc.getPlayer(iCiv).initUnit(iHeavySwordsman, x, y, UnitAITypes.UNITAI_ATTACK, DirectionTypes.DIRECTION_SOUTH)
+								gc.getPlayer(iCiv).initUnit(iHeavySwordsman, x, y, UnitAITypes.UNITAI_ATTACK_CITY, DirectionTypes.DIRECTION_SOUTH)
 
 						else:
-							gc.getPlayer(iCiv).initUnit(iSwordsman, tPlot[0], tPlot[1], UnitAITypes.UNITAI_ATTACK, DirectionTypes.DIRECTION_SOUTH)
-							gc.getPlayer(iCiv).initUnit(iSwordsman, tPlot[0], tPlot[1], UnitAITypes.UNITAI_ATTACK_CITY, DirectionTypes.DIRECTION_SOUTH)
+							gc.getPlayer(iCiv).initUnit(iSwordsman, x, y, UnitAITypes.UNITAI_ATTACK, DirectionTypes.DIRECTION_SOUTH)
+							gc.getPlayer(iCiv).initUnit(iSwordsman, x, y, UnitAITypes.UNITAI_ATTACK_CITY, DirectionTypes.DIRECTION_SOUTH)
 
 	def giveEarlyColonists( self, iCiv):
 		pCiv = gc.getPlayer(iCiv)
 		teamCiv = gc.getTeam(pCiv.getTeam())
-		if (pCiv.isAlive() and pCiv.isHuman() == False):
+		if pCiv.isAlive() and not pCiv.isHuman():
 			capital = gc.getPlayer(iCiv).getCapitalCity()
 			tCapital = (capital.getX(), capital.getY())
 			tSeaPlot = self.findSeaPlots(tCapital, 1, iCiv)
 			
-			if iCiv == iRome:
-				cityList = PyPlayer(iCiv).getCityList()
-				for city in cityList:
-					pCity = city.GetCy()
-					if pCity.getRegionID() == rIberia:
-						tCapital = (pCity.getX(), pCity.getY())
-						break
+			# if iCiv == iRome:

```

Comment:
The only thing that changes is tCapital. But tCapital is not used anymore in that function. I assume it should be before tSeaPlot = self.findSeaPlots(tCapital, 1, iCiv), so it is used.